agent/consul/state/acl_events.go

```diff
@@ -0,0 +1,73 @@
+package state
+
+import (
+	"github.com/hashicorp/consul/agent/consul/state/db"
+	"github.com/hashicorp/consul/agent/consul/stream"
+	"github.com/hashicorp/consul/agent/structs"
+	memdb "github.com/hashicorp/go-memdb"
+)
+
+// aclChangeUnsubscribeEvent creates and returns stream.UnsubscribeEvents that
+// are used to unsubscribe any subscriptions which match the tokens from the events.
+//
+// These are special events that will never be returned to a subscriber.
+func aclChangeUnsubscribeEvent(tx db.ReadTxn, changes db.Changes) ([]stream.Event, error) {
```

Comment:
This has changed quite a bit from previous commits. Previously it produced regular events that were not available to subscriptions. Now this is specifically for internal ACL events. In the future when we add ACL events for subscriptions we can add a different handler.